core/metrics-spi/src/main/java/software/amazon/awssdk/metrics/provider/DefaultMetricConfigurationProvider.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.metrics.provider;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.metrics.MetricCategory;
+
+/**
+ * The default implementation of {@link MetricConfigurationProvider} interface.
+ */
+@SdkPublicApi
+public final class DefaultMetricConfigurationProvider implements MetricConfigurationProvider {
+
+    private final boolean enabled;
+    private final Set<MetricCategory> metricCategories;
+
+    private DefaultMetricConfigurationProvider(Builder builder) {
+        this.enabled = builder.enabled;
+        this.metricCategories = Collections.unmodifiableSet(resolveCategories(builder.categories));
+    }
+
+    @Override
+    public boolean enabled() {
+        return enabled;
+    }
+
+    @Override
+    public Set<MetricCategory> metricCategories() {
+        return metricCategories;
+    }
+
+    /**
+     * @return a new {@link Builder} instance
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    private Set<MetricCategory> resolveCategories(Set<MetricCategory> categories) {
+        if (categories == null || categories.isEmpty()) {
+            return new HashSet<>(Arrays.asList(MetricCategory.Default));
```

Comment:
Is this the behavior we want?

Reply:
Yes, it doesn't make sense to not have a category if metrics feature is enabled. So SDK defaults to DEFAULT